bluecellulab/tools.py

```diff
@@ -110,51 +113,92 @@ def load_nrnmechanisms(libnrnmech_path: str) -> None:
+    morphology_path: str | Path,
+    template_format: str,
+    emodel_properties: EmodelProperties | None,
+    current_delta: float = 0.01,
+) -> float:
     """Calculate the input resistance at rest of the cell."""
-    rest_voltage = calculate_SS_voltage(template_name, morphology_name, 0.0)
+    rest_voltage = calculate_SS_voltage(
+        template_path, morphology_path, template_format, emodel_properties, 0.0
+    )
     step_voltage = calculate_SS_voltage(
-        template_name, morphology_name, current_delta)
+        template_path,
+        morphology_path,
+        template_format,
+        emodel_properties,
+        current_delta,
+    )
 
     voltage_delta = step_voltage - rest_voltage
 
     return voltage_delta / current_delta
 
 
-def calculate_SS_voltage(template_name, morphology_name, step_level):
-    """Calculate the steady state voltage at a certain current step."""
+def calculate_SS_voltage(
+    template_path: str | Path,
+    morphology_path: str | Path,
+    template_format: str,
+    emodel_properties: EmodelProperties | None,
+    step_level: float,
+) -> float:
+    """Calculate the steady state voltage at a certain current step.
+
+    The use of Pool is safe here since it will just run a single task.
+    """
     pool = multiprocessing.Pool(processes=1)
     SS_voltage = pool.apply(
-        calculate_SS_voltage_subprocess, [
-            template_name, morphology_name, step_level])
+        calculate_SS_voltage_subprocess,
+        [
+            template_path,
+            morphology_path,
+            template_format,
+            emodel_properties,
+            step_level,
+        ],
+    )
     pool.terminate()
     return SS_voltage
 
 
-def calculate_SS_voltage_subprocess(template_name, morphology_name,
-                                    step_level, check_for_spiking=False,
-                                    spike_threshold=-20.0):
+def calculate_SS_voltage_subprocess(
+    template_path: str | Path,
+    morphology_path: str | Path,
+    template_format: str,
+    emodel_properties: EmodelProperties | None,
+    step_level: float,
+    check_for_spiking=False,
+    spike_threshold=-20.0,
+) -> float:
     """Subprocess wrapper of calculate_SS_voltage.
 
-    if check_for_spiking is True, this function will return None if the
-    cell spikes from from 100ms to the end of the simulation indicating
-    no steady state was reached.
+    This code should be run in a separate process. If check_for_spiking
+    is True, this function will return None if the cell spikes from
+    100ms to the end of the simulation indicating no steady state was
+    reached.
     """
-    cell = bluecellulab.Cell(template_name, morphology_name)
+    cell = bluecellulab.Cell(
+        template_path=template_path,
+        morphology_path=morphology_path,
+        template_format=template_format,
+        emodel_properties=emodel_properties,
+    )
     cell.add_ramp(500, 5000, step_level, step_level)
```

Comment:
Do we usually use the ramp protocol for this kind of feature?

Reply:
Good point. I had the same question. This is an old code that we did not use for a while.

In this PR I only wanted to add the tests to make sure the function will support the Sonata format and the open-sourced SSCX circuit. I didn't want to touch the protocols.

How about we create an issue from this line and ask Werner why ramp is used here when he's back?

Reply:
alright, sounds good

Reply:
Here it is: https://github.com/BlueBrain/BlueCelluLab/issues/77

Reply:
thanks!